src/Swift.Runtime/src/Swift/SwiftEquatable.cs

```diff
@@ -40,31 +40,21 @@ public static unsafe bool Equals<T>(T lhs, T rhs)
             var metadata = TypeMetadata.GetTypeMetadataOrThrow<T>();
             var equatablePwt = ProtocolWitnessTable.GetOrThrow<T, IEquatable<T>>();
 
-            IntPtr lhsPayload = IntPtr.Zero;
-            IntPtr rhsPayload = IntPtr.Zero;
+            var lhsPayloadSpan = stackalloc byte[(int)metadata.Size];
+            IntPtr lhsPayload = (IntPtr)Unsafe.AsPointer(ref lhsPayloadSpan[0]);
```

Comment:
```suggestion
            byte* lhsPayload = stackalloc byte[(int)metadata.Size];
```
Since you are not using the Span anyway, you can skip the ceremony and allocate `byte*` directly.

Reply:
(I am not commenting on all places.)